src/index.ts

```diff
@@ -13,49 +13,100 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import * as extend from 'extend';
 import * as path from 'path';
-import {AuthenticationConfig, Common, ServiceConfig} from '../third_party/types/common-types';
+import * as request from 'request';
+
+import {AuthenticationConfig, Common, ServiceConfig, ServiceObject} from '../third_party/types/common-types';
+
 import {Config, defaultConfig} from './config';
 import {Profiler, ProfilerConfig} from './profiler';
 
 const common: Common = require('@google-cloud/common');
+const metadataAPI = 'http://metadata.google.internal/computeMetadata/v1/';
+
+// Returns true if running on GCE and false otherwise.
+// It is assumed a program is running on GCE if the metadata API can be
+// accessed.
+function onGCE(): Promise<boolean> {
+  return new Promise<boolean>((resolve) => {
+    request.get(
+        {
+          url: metadataAPI,
+          timeout: 2000,
+          headers: {'Metadata-Flavor': 'Google'}
+        },
+        function(err: Error, response) {
+          err ? resolve(false) : resolve(true);
+        });
+  });
+}
+
+// Returns value of metadata field.
+// Throws error if there is a problem accessing metadata API.
+function getField(field: string): Promise<string> {
```

Comment:
This functionality is provided by gcp-metadata.

Reply:
I'm a bit concerned about how slow gcp-metadata is. 
It took about  6 seconds until it threw an error when I ran it locally, and I don't see a clear way to make it faster.